modules/frontend/metrics_query_range_sharder.go

```diff
@@ -115,6 +116,7 @@ func (s queryRangeSharder) RoundTrip(pipelineRequest pipeline.Request) (pipeline
 		}
 	}
 	req.Exemplars = maxExemplars
+	req.MaxSeries = uint32(s.cfg.MaxResponseSeries)
```

Comment:
i find the code that is setting the value on the request confusing. we are doing it in the handler, again here, and in the backend requests below. i'm finding this confusing.

Reply:
the backend request at the bottom is wrong, but this is necessary because in the handler, the value is only passed to the combiner https://github.com/grafana/tempo/blob/ba3fb50572d413af790a74a06d3f3cbd4adc6f76/modules/frontend/metrics_query_range_handler.go#L93 

but the http req was never modified before being passed to the sharder https://github.com/grafana/tempo/blob/ba3fb50572d413af790a74a06d3f3cbd4adc6f76/modules/frontend/metrics_query_range_handler.go#L118

Reply:
i'm good on this 👍 

i just realized i basically put this comment 2x.